bitclock-fw/main/libs/mqtt.c

```diff
@@ -0,0 +1,100 @@
+#include "esp_log.h"
+#include "mqtt_client.h"
+#include "mqtt.h"
+
+static const char *TAG = "mqtt";
+
+esp_mqtt_client_handle_t client;
+static bool configs_published = false;
+
+static const char *co2_unique_id = "bitclock_co2";
+static const char *nox_unique_id = "bitclock_nox";
+static const char *voc_unique_id = "bitclock_voc";
+static const char *temp_unique_id = "bitclock_temp";
+static const char *humidity_unique_id = "bitclock_humidity";
+static char config_message[300];
+static char config_topic[100];
+static char state_topic[100];
+
+bool publish_message(const char *topic, const char *message, bool retain) {
+    bool success = esp_mqtt_client_publish(client, topic, message, 0, 1, retain) >= 0;
+    if (success) {
+      ESP_LOGI(TAG, "MQTT publish successful, topic=%s message=%s", topic, message);
+    } else { 
+      ESP_LOGE(TAG, "MQTT publish failed, topic=%s message=%s", topic, message);
+    }
+    return success;
+}
+
+bool publish_homeassistant_config(const char * name, const char * unique_id, const char * device_class, const char * unit_of_measurement, const char * state_class, const char * platform) {
+    snprintf(config_message, "{\"name\": \"%s\",\"unique_id\": \"%s\", \"device_class\": \"%s\", \"unit_of_measurement\": \"%s\",\"state_class\": \"%s\",\"platform\": \"%s\"}", name, unique_id, device_class, unit_of_measurement, state_class, platform);
+    snprintf(config_topic, "homeassistant/sensor/%s/config", unique_id);
+    return publish_message(config_topic, config_message, true);
+}
+
+bool publish_homeassistant_sensor(const char * unique_id, const char * state_message) {
+    snprintf(state_topic, "homeassistant/sensor/%s/state", unique_id);
```

Comment:
the use of snprintf looks off.  check out snprintf params:
https://www.geeksforgeeks.org/snprintf-c-library/

It needs to include the max size of the output buffer